src/events/reactionAdd.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  Client,
+  EmbedBuilder,
+  Events,
+  ForumChannel,
+  MessageReaction,
+  User,
+} from 'discord.js';
+import ANONY_CONSTANTS from '../constants/anony';
+
+const reactionAddEvent = {
+  name: Events.MessageReactionAdd,
+  async execute(client: Client, reaction: MessageReaction, user: User) {
+    if (user.bot) return;
+    if (reaction.partial) {
+      try {
+        await reaction.fetch();
+      } catch (error) {
+        console.error(
+          'Something went wrong when fetching the message: ',
+          error,
+        );
+        return;
+      }
+    }
+
+    if (
+      reaction.message.channel.id !== process.env.ANONYMOUS_APPROVAL_CHANNEL_ID
+    )
+      return;
+    if (reaction.message.embeds.length !== 1) return;
+    if (reaction.message.embeds[0].fields.length === 2) return; // Because reacted submission will have one extra field which is approved or rejected
```

Comment:
IMHO, will it be better if we use `embes[0].color` to distinguish for this statement?

Reply:
You're right